types/linkifyjs/index.d.ts

```diff
@@ -13,7 +13,9 @@ export type PossiblyFuncOfHrefAndType<T> =
 
 // There's always a possibility that values at a string key can always be
 // undefined; i.e., if the key does not exist.
-export type PossiblyByType<T> = T | { [type: string]: T | undefined };
+export type PossiblyByType<T> = T | { [type: string]: T | React.ReactElement | undefined };
+
+export type FormatFunc = (value: string, type: string) => string | React.ReactElement;
```

Comment:
Can you test whether returning `undefined`/`null`/`number`/an array of the above is allowed? If so, the return type can simply be `React.ReactNode`.

Reply:
Good idea! I just tried it, using `export type FormatFunc = (value: string, type: string) => React.ReactNode;` in my type declaration in my project. Returning `null` or `undefined` works and results in an `<a>` tag with no children, as you might expect. Returning a number works and results in that number being the clickable text of the link. Returning an array also works; I tried returning `['cat', 'dog']` and the link text became `catdog`.